multi/multi.go

```diff
@@ -156,6 +187,36 @@ func (connMulti *ConnectionMulti) checker() {
 	}
 }
 
+func (connMulti *ConnectionMulti) refreshNodes() {
+	for connMulti.getState() != connClosed {
+		refreshTimer := time.NewTimer(connMulti.opts.ClusterDiscoveryTime)
+		select {
+		case <-refreshTimer.C:
+			if connMulti.getState() == connClosed || len(connMulti.opts.NodesGetFunctionName) <= 0 {
+				continue
+			}
+			resp, err := connMulti.Call(connMulti.opts.NodesGetFunctionName, []interface{}{})
+			if err != nil {
+				continue
+			}
+			if len(resp.Data) > 0 {
+				data := resp.Data[0].([]interface{})
+				addrs := make([]string, len(data))
+				for i, v := range data {
+					addrs[i] = v.(string)
+				}
+				connMulti.addrs = addrs
+				connMulti.clearConnectionPool()
+			}
+
+			somebodyAlive, _ := connMulti.warmUp()
```

Comment:
So, you really want to drop all connections and reconnect every 60 seconds?

Reply:
Reworked to keep actual connections in pool and clean only obsolete